rskj-core/src/main/java/co/rsk/cli/tools/ShowInfo.java

```diff
@@ -0,0 +1,150 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.cli.tools;
+
+import co.rsk.cli.PicoCliToolRskContextAware;
+import co.rsk.trie.Trie;
+import co.rsk.trie.TrieStore;
+import org.ethereum.core.Block;
+import org.ethereum.db.BlockStore;
+import org.spongycastle.util.encoders.Hex;
+import picocli.CommandLine;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * The entry point for show info tool.
+ */
+@CommandLine.Command(name = "showinfo", mixinStandardHelpOptions = true, version = "showinfo 1.0",
+        description = "The entry point for show info tool")
+public class ShowInfo extends PicoCliToolRskContextAware {
+    @CommandLine.Option(names = {"-cmd", "--command"}, description = "Commands:\n\n" +
+            "STATEROOT - shows state root info.\n" +
+            "TRIESTORE - shows trie store info. \n" +
+            "STATESIZES - shows state sizes info. \n" +
+            "DB - shows db info. \n", required = true)
+    private String command;
+
+    @CommandLine.Option(names = {"-b", "--block"}, description = "Block number")
+    private Long blockNumber;
+
+    enum Command {
+        STATEROOT("STATEROOT"),
+        TRIESTORE("TRIESTORE"),
+        STATESIZES("STATESIZES"),
+        DB("DB");
+
+        private final String name;
+
+        Command(@Nonnull String name) {
+            this.name = name;
+        }
+
+        public static Command ofName(@Nonnull String name) {
+            Objects.requireNonNull(name, "name cannot be null");
+            return Arrays.stream(Command.values()).filter(cmd -> cmd.name.equals(name) || cmd.name().equals(name))
+                    .findAny()
+                    .orElseThrow(() -> new IllegalArgumentException(String.format("%s: not found as Command", name)));
+        }
+    }
+
+    public static void main(String[] args) {
+        create(MethodHandles.lookup().lookupClass()).execute(args);
+    }
+
+    @Override
+    public Integer call() throws IOException {
+        Command command = Command.ofName(this.command.toUpperCase(Locale.ROOT));
+
+        BlockStore blockStore = ctx.getBlockStore();
+        TrieStore trieStore;
+        long blockNumber = -1;
```

Comment:
This overwrites the block number passed through command line option  `Long blockNumber` in L50. Leads to an error for `-cmd STATEROOT`. Perhaps remove this line and just use `long` instead of `Long` in L50? Or some other fix. Thanks.

Reply:
Addressed